src/devices/machine/vic_pl192.cpp

```diff
@@ -0,0 +1,227 @@
+// license:BSD-3-Clause
+// copyright-holders:Melissa Goad
+
+// ARM PrimeCell PL192 VIC emulation
+
+#include "emu.h"
+#include "machine/vic_pl192.h"
+
+#define LOG_GENERAL (1U << 0)
+
+#define VERBOSE (LOG_GENERAL)
+#include "logmacro.h"
+
+void vic_pl192_device::set_irq_line(int irq, int state)
+{
+	u32 mask = (1 << irq);
+
+    if(state)
+    {
+        raw_intr |= mask;
+    }
+    else
+    {
+        raw_intr &= ~mask;
+    }
+
+    if(u32 intrs = (raw_intr | soft_intr) & intr_en)
+    {
+        if(intrs & intr_select) m_out_fiq_func(1);
+        else m_out_fiq_func(0);
+
+        if(intrs & ~intr_select) m_out_irq_func(1);
+        else m_out_irq_func(0);
+    }
+}
+
+uint32_t vic_pl192_device::read(offs_t offset)
+{
+    offset &= 0xffc;
+    switch(offset)
+    {
+        case 0x000:
```

Comment:
Device should be a device_memory_interface with these offsets into an address_map. Will save some typing and indentation.

Reply:
I honestly don't see how that would save on typing, but okay...

Reply:
Oh well, I did it your way as requested.